lib/index.js

```diff
@@ -0,0 +1,126 @@
+  const formattedPublishedLabels = await labelFormatter(labelData);
+  const data = fs.readFileSync('./label-config.json', { encoding: 'utf8' });
+
+  const localConfig = JSON.parse(data.toString());
+  const illegalLabels = checkForSameLabelName(
+    localConfig,
+    formattedPublishedLabels
+  );
+  if (illegalLabels) {
+    throw new Error(
+      'The config has labels with the same name. The Label names have to be unique.'
+    );
+  }
+  const newLabels = checkForNewOrOldLabels(
+    localConfig,
+    formattedPublishedLabels
+  );
+  const labelsToDelete = checkForNewOrOldLabels(
+    formattedPublishedLabels,
+    localConfig
+  );
+  const labelsToUpdate = checkForModifications(
+    localConfig,
+    formattedPublishedLabels
+  );
+  if (newLabels.length !== 0) {
+    await createLabels(newLabels);
+  }
+  if (labelsToDelete.length !== 0) {
+    await deleteLabels(labelsToDelete);
+  }
+  if (labelsToUpdate.length !== 0) {
+    await updateLabels(labelsToUpdate);
+  }
+  await writeLabelsConfig();
+  const changes =
+    newLabels.length + labelsToDelete.length + labelsToUpdate.length;
+  const summary =
+    changes !== 0
+      ? {
+          message: 'Labels synchronized!',
+          stats: {
+            created: newLabels.length,
+            deleted: labelsToDelete.length,
+            updated: labelsToUpdate.length,
+          },
+        }
+      : { message: 'Labels already up to date!' };
+  return summary;
+}
+
+async function writeLabelsConfig() {
+  const labelData = await fetchLabels();
+  const labels = await labelFormatter(labelData);
+  fs.writeFileSync('./label-config.json', JSON.stringify(labels, null, 2));
+}
+
+function checkForNewOrOldLabels(newLabels, oldLabels) {
+  const updatedLabels = [];
+  const OldLabelIds = [];
+  oldLabels.forEach(label => {
+    OldLabelIds.push(label.id ? label.id : undefined);
+  });
+  newLabels.forEach(label => {
+    if (!OldLabelIds.includes(label.id ? label.id : undefined)) {
+      updatedLabels.push(label);
+    }
+  });
+  return updatedLabels;
+}
+
+function checkForModifications(localConfig, publishedConfig) {
+  const formattedPublishedConfig = labelFormatter(publishedConfig);
+  const labelChanges = [];
+  localConfig.forEach(localLabel => {
+    formattedPublishedConfig.forEach(publicLabel => {
+      if (localLabel.id === publicLabel.id) {
+        if (
+          Object.values(localLabel).toString() !==
+          Object.values(publicLabel).toString()
+        ) {
+          labelChanges.push(localLabel);
+        }
+      }
+    });
+  });
+  return labelChanges;
+}
+
+function checkForSameLabelName(localConfig, publishedConfig) {
+  const illegalLabels = localConfig.map(localLabel => {
+    const illegalname = publishedConfig.find(publicLabel => {
+      return localLabel.name === publicLabel.name ? publicLabel : undefined;
+    });
+    return illegalname;
+  });
+  return illegalLabels;
+}
+
+function labelFormatter(labelData) {
```

Comment:
I think it makes more sense to do the mapping in the `fetchLabels` function.

Reply:
True...
cb03194
🥇